### PR TITLE
Add mitmproxy_wireguard

### PR DESCRIPTION
**Pull request by AdrianFreundQC**
This is a new dependency for mitmproxy v9.0.0

mitmproxy feedstock PR: https://github.com/conda-forge/mitmproxy-feedstock/pull/10
mitmproxy v9.0.0 changelog: https://github.com/mitmproxy/mitmproxy/compare/v8.1.1...9.0.0?diff=split#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R86

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by freundTech** [comment]
I'm OK with being listed as a maintainer.

**Comment by AdrianFreundQC** [comment]
@conda-forge/python 
@conda-forge/staged-recipes 
This should be ready for review.

File: recipes/mitmproxy_wireguard/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: True  # [py<37]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - rust
```
Comment (xhochy) [inline recipes/mitmproxy_wireguard/meta.yaml:20]:
```suggestion
    - {{ compiler('rust' }}
```

File: recipes/mitmproxy_wireguard/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - maturin
```
Comment (xhochy) [inline recipes/mitmproxy_wireguard/meta.yaml:21]:
```suggestion
    - python                                 # [build_platform != target_platform]
    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
    - maturin                                # [build_platform != target_platform]
```

**Review by xhochy — APPROVED** [review]

**Comment by xhochy** [comment]
> `error: failed to run the Python interpreter at C:\bld\mitmproxy_wireguard_1667838681835\_build_env\python.exe:`

This error looks like `maturin` is looking in the wrong place for the Python compiler. I'm a bit confused as this is not happening in other recipes.

**Comment by AdrianFreundQC** [comment]
Thanks for the fix!

**Pull request merged**